news/2020-02-03-Micro_Profile_OpenTracing_Comes_To_WildFly.adoc

```diff
@@ -0,0 +1,123 @@
+:awestruct-layout: blog
+:source-highlighter: coderay
+:encoding: utf-8
+:lang: en
+:awestruct-description: How to configure and use Micro Profile OpenTracing within WildFly
+
+Until WildFly 19 you could use MPOT (Micro Profile OpenTracing) to trace your application using environment variables. With wildFly 19 you can now configure several Jaeger Tracers to be used in your applications.
+
+== Installing Jaeger
+
+Let's start a jaeger instance using docker :
+[source,bash]
+--
+docker run -d --name jaeger \
+  -p 16831:6831/udp \
+  -p 5778:5778 \
+  -p 14268:14268 \
+  -p 16686:16686 \
+  jaegertracing/all-in-one:1.16
+--
+
+Now, you can navigate to link:http://localhost:16686[http://localhost:16686] to access the Jaeger UI.
+
+[NOTE]
+====
+For demonstration purpose we have mapped the port *6831* to *16831*.
+====
+
+== Configuring the OpenTracing subsystem
+
+You can use either UDP or TCP to send your log spans to Jaeger.  Note that those configurations are *exclusive*, so if yo configure an TCP endpoint then Jaeger won't use the UDP configuration.
+
+=== Using UDP
+
+First we define an outbound socket binding towards the Jaeger tracer.
+[source,ruby]
+--
+[standalone@localhost:9990 /] /socket-binding-group=standard-sockets/remote-destination-outbound-socket-binding=jaeger:add(host=localhost, port=16831)
+{"outcome" => "success"}
+--
+
+Now we can define our MPOT tracer configuration:
+[source,ruby]
+--
+[standalone@localhost:9990 /] /subsystem=microprofile-opentracing-smallrye/jaeger-tracer=jaeger-demo:add(sampler-type=const, sampler-param=1, reporter-log-spans=true, sender-binding=jaeger)
+{"outcome" => "success"}
+--
+
+=== Using TCP
+
+If you want to use TCP instead of UDP you need to configure the *sender-endpoint* and set its value to *http://localhost:14268/api/traces*.
+
+[source,ruby]
+--
+[standalone@localhost:9990 /] /subsystem=microprofile-opentracing-smallrye/jaeger-tracer=jaeger:write-attribute(name="sender-endpoint", value="http://localhost:14268/api/traces")
+{
+    "outcome" => "success",
+    "response-headers" => {
+        "operation-requires-reload" => true,
+        "process-state" => "reload-required"
+    }
+}
+[standalone@localhost:9990 /] :reload
+{
+    "outcome" => "success",
+    "result" => undefined
+}
+--
+
+=== Setting the default tracer
+
+Let's define this new tracer as the default tracer to be used by WildFly:
+[source,ruby]
+--
+[standalone@localhost:9990 /] /subsystem=microprofile-opentracing-smallrye:write-attribute(name=default-tracer, value=jaeger-demo)
+{
+    "outcome" => "success",
+    "response-headers" => {
+        "operation-requires-reload" => true,
+        "process-state" => "reload-required"
+    }
+}
+[standalone@localhost:9990 /] :reload
+{
+    "outcome" => "success",
+    "result" => undefined
+}
+--
+
+[NOTE]
+====
+You can use the *web.xml* deployment descriptor and set the init-parameter *smallrye.opentracing.tracer.configuration* to the name of tracer to use.
```

Comment:
s/can use/can alternatively use/g

or something like that; i.e. AIUI you are describing an alternate approach.